### ISSUE TITLE
Could not go to Augmentation.

### ISSUE THREAD

**Issue by b14cknc0d3** [Dart-Code/Dart-Code#5112]
**Describe the bug**
Could not go to Augmentation
**To Reproduce**
simple code
```
@JsonCodable()
class Person {
  final String name;
  final int age;

  Person({required this.name, required this.age});
}

void main(List<String> args) {
  final person = Person(name: 'John', age: 30);
  print(person.toJson());
}
```
when I run  this `dart run --enable-experiment=macros lib/dart_marcros.dart` 
I got this `{name: John, age: 30}` result but I can't go to augmentation.

**Expected behavior**
I should able to go to augmentation  either by clicking `Go to augmentation`  or `toJson()`.
**Screenshots**

**Please complete the following information:**
When I click go to augmentation I got this error.
```
file:///Users/***/Documents/Playgrounds/dart_marcros/lib/dart_marcros.macro.dart. Detail: Unable to read file '/Users/***/Documents/Playgrounds/dart_marcros/lib/dart_marcros.macro.dart' (Error: Unable to resolve nonexistent file '/Users/***/Documents/Playgrounds/dart_marcros/lib/dart_marcros.macro.dart')
```

You can run the **Dart: Collect Diagnostic Information** command from the VS Code command palette (`F1`) to easily capture this information or provide it manually.

- Dart Code extension: 3.88.0
- Flutter extension: 3.88.0 (not activated)
- App: VSCodium
- App Host: desktop
- Version: mac 1.88.1
- Workspace type: Dart (LSP)
- Dart (3.5.0-167.0.dev)

**Comment by DanTup** [comment]
@b14cknc0d3 please update to the latest version of the VS Code extension (v3.88.1) and the issue should be resolved. Before v3.88.1 you would need to set a VS Code setting (`dart.experimentalMacroSupport`) to enable IDE support for generated macro sources, but with 3.88.1 this is enabled automatically for v3.5 Dart SDKs.

**Comment by petrnymsa** [comment]
> @b14cknc0d3 please update to the latest version of the VS Code extension (v3.88.1) and the issue should be resolved. Before v3.88.1 you would need to set a VS Code setting (`dart.experimentalMacroSupport`) to enable IDE support for generated macro sources, but with 3.88.1 this is enabled automatically for v3.5 Dart SDKs.

Happening same to me. Extension version 3.88.1. 

Dart version, OS windows. 
```dart
Dart SDK version: 3.5.0-167.0.dev (dev) (Fri May 17 05:05:43 2024 -0700) on "windows_x64"
```

Go to Augmentation disappears. Only Develooper: Reload window helps.

**Comment by DanTup** [comment]
If you're seeing it disappear and come back after a reload, that suggests something may be getting into a bad state in the analysis server - if you can provide steps to reproduce or an [instrumentation log](https://dartcode.org/docs/logging/#analyzer-instrumentation), please open a new issue with the details. Thanks!

**Comment by b14cknc0d3** [comment]
Hello. After restart my Laptop I can go to Augmentation but I don't set this `dart.experimentalMacroSupport`  in VSCodium setting and version is `3.88.0` (that is the latest I can update in VSCodium.)

**Comment by DanTup** [comment]
v3.88.1 was only published to VS Code, so if you're using something that uses the extension from openvsx and only have v3.88.0 then you'll need to add:

```js
"dart.experimentalMacroSupport": true,
```

To your settings for now. It will say that the setting is not recognised, but just ignore this, it will work it's just not a declated setting that Code will recognise. This should fix navigation errors showing errors for ".macro.dart" files and will instead navigate to the generated sources.

**Comment by DanTup** [comment]
I've published v3.88.1 to open-vsx, so it should show up soon for things like Codium and anything else unable to use the official VS Code marketplace. With that version, the setting above is unnecessary.
